src/app/components/elements/views/ActivityGraph.tsx

```diff
@@ -1,17 +1,24 @@
-import React, { useEffect } from "react";
-import { bb } from "billboard.js";
+import React, { useEffect, useMemo } from "react";
+import bb, { zoom, areaSpline } from "billboard.js";
 import { AnsweredQuestionsByDate } from "../../../../IsaacApiTypes";
 import { formatISODateOnly } from "../DateString";
 
-export const ActivityGraph = ({ answeredQuestionsByDate }: { answeredQuestionsByDate: AnsweredQuestionsByDate }) => {
-  let selectedDates: string[] = [];
+const filterSelectedDates = (answeredQuestionsByDate: AnsweredQuestionsByDate) => {
```

Comment:
The name of this function may need rethinking, because it also sorts them.

Reply:
Renamed to `filterAndSortSelectedDates`